File: pootle/static/js/shared/utils/fetch.js
/*
 * Copyright (C) Pootle contributors.
 *
 * This file is a part of the Pootle project. It is distributed under the GPL3
 * or later license. See the LICENSE file for a copy of the license and the
 * AUTHORS file for copyright and authorship information.
 */

import $ from 'jquery';


const requests = {};


function fetch({ url, body, method = 'GET', dataType = 'json', queue = null,
                 crossDomain = false }) {
  const queueName = queue || url;

  if (requests[queueName]) {
    requests[queueName].abort();
  }

  requests[queueName] = (
    $.ajax({
      crossDomain,
      method,
      dataType,
      url: l(url),
      data: body,
    })
  );
  requests[queueName].done(() => {requests[queueName] = null;});
  return requests[queueName];
}


export default fetch;

File: pootle/apps/pootle_fs/management/commands/fs_commands/fetch.py
# -*- coding: utf-8 -*-
#
# Copyright (C) Pootle contributors.
#
# This file is a part of the Pootle project. It is distributed under the GPL3
# or later license. See the LICENSE file for a copy of the license and the
# AUTHORS file for copyright and authorship information.

from pootle_fs.management.commands import ProjectSubCommand


class FetchCommand(ProjectSubCommand):
    help = "Fetch remote filesystems."

    def handle(self, **options):
        self.get_fs(options["project"]).fetch()
